File: porjar-api/internal/model/news.go
package model

import "time"

type News struct {
	ID          int64     `json:"id" db:"id"`
	Title       string    `json:"title" db:"title"`
	Slug        string    `json:"slug" db:"slug"`
	Excerpt     string    `json:"excerpt" db:"excerpt"`
	Content     string    `json:"content,omitempty" db:"content"`
	Category    string    `json:"category" db:"category"`
	ImageURL    *string   `json:"image_url,omitempty" db:"image_url"`
	PublishedAt time.Time `json:"published_at" db:"published_at"`
	CreatedAt   time.Time `json:"created_at" db:"created_at"`
	UpdatedAt   time.Time `json:"updated_at" db:"updated_at"`
}

File: porjar-web/src/types/news.ts
export interface News {
  id: number
  title: string
  slug: string
  excerpt: string
  content: string
  category: string
  image_url?: string | null
  published_at: string
  created_at: string
  updated_at: string
}

export interface NewsFormData {
  title: string
  slug: string
  excerpt: string
  content: string
  category: string
  image_url: string
  published_at: string
}

export const NEWS_CATEGORIES = ['Berita', 'Pengumuman', 'Event', 'Hasil'] as const
